mentoring/mentoring.py

```diff
@@ -144,14 +177,31 @@ def student_view(self, context):
         # Migrate stored data if necessary
         self.migrate_fields()
 
-        fragment, named_children = self.get_children_fragment(
-            context, view_name='mentoring_view',
-            not_instance_of=self.FLOATING_BLOCKS,
-        )
+        title = ""
+        header = ""
+
+        fragment = Fragment()
+        child_content = u""
+        for child_id in self.children:
+            child = self.runtime.get_block(child_id)
+            if isinstance(child, TitleBlock):
```

Comment:
@bradenmacdonald oh, you redecorated it... I don't like it. :) Being serious - I don't like this loop. I understand what's happening here, and it should just work, but it's a bit clumsy. Can't we first separate special-case blocks into their own list(s), than only process usual ones like this, and give special treatment to specials? Something like

    title_blocks, header_blocks, messages_blocks, ordinary_blocks = self._split_children()
    # _split_children is probably going to do self.runtime.get_block(child_id) internally, than return actual blocks
    for child_block in ordinary_blocks:
       ....
    title = title_blocks[-1].content
    header = header_blocks[-1].render(...)

Note I'm using `-1` to get last block just to preserve the behavior this code exhibits now - later blocks overwrite former ones. It might be more than that, e.g. asserting that there are only one such block, etc.